src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,175 @@
+package duke;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.ArrayList;
+import java.util.List;
+
+import duke.exception.CorruptedStorageException;
+import duke.exception.InvalidDateInputException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+
+/**
+ * A class that represents the storage of the data
+ */
+public class Storage {
+
+    private final Path filePath;
+    private List<String> serialisedTasks;
+
+    /**
+     * Initialises a new instance.
+     *
+     * @param filePath The path to save the data to in the hard disk.
+     */
+    public Storage(Path filePath) {
+        this.filePath = filePath;
+
+        try {
+            // Create directories if it does yet exist
+            Path parentPath = filePath.getParent();
+            Files.createDirectories(parentPath);
+
+            // Check if the file to be read exists. If not, create it.
+            if (!Files.exists(filePath)) {
```

Comment:
Perhaps this is a redundant check, as per the [documentation of the createFile method](https://docs.oracle.com/javase/7/docs/api/java/io/File.html#createNewFile())?

Reply:
Thanks for pointing it out. Unfortunately, if I remove this check my program doesn't seem to run when there's an existing file already. Not sure why too.